src/Controller/AppController.php

```diff
@@ -168,9 +168,40 @@ protected function prepareRequest($type) : array
             $data['api'] = $api;
         }
 
+        // prepare attributes: only modified attributes
+        if (!empty($data['actualAttributes'])) {
+            $attributes = json_decode($data['actualAttributes'], true);
+            foreach ($attributes as $key => $value) {
+                // remove unchanged attributes from $data
+                if (isset($data[$key]) && !$this->hasChanged($value, $data[$key])) {
+                    unset($data[$key]);
+                }
+            }
+            unset($data['actualAttributes']);
+        }
+
         return $data;
     }
 
+    /**
+     * Return true if $value1 equals $value2 or both are empty (null|'')
+     *
+     * @param string|array $value1 The first value | field value in model data (db)
+     * @param string|array $value2 The second value | field value from form
+     * @return bool
+     */
+    private function hasChanged($value1, $value2)
+    {
+        if (is_bool($value1) && !is_bool($value2)) { // i.e. true / "1"
+            return $value1 !== boolval($value2);
+        }
+        if (empty($value1) && empty($value2)) { // i.e. null / ""
```

Comment:
I would use a different condition like:
```if (($value1 === null || $value1 === '') && ($value2 === null || $value2 === ''))```

Reason: `0` returns true with `empty()` 
Can we have a `null` value from a field input? :thinking: 
Probably other chenges are necessary.

But we absolutely need some unit tests here